anya/roman_numeral.rb

```diff
@@ -0,0 +1,36 @@
+class RomanNumeral
+  ROMAN_NUM = {
+              'M' => 1000,
+              'D' => 500,
+              'C' => 100,
+              'L' => 50,
+              'X' => 10,
+              'V' => 5,
+              'I' => 1
+            }
+  ROMAN_NUM.default = 0
+
+  def initialize(number)
+    @number = number.respond_to?(:upcase) ? number.upcase : ""
+    valid?
```

Comment:
I think it's better design to use the "tell, don't ask" principle here. So for e.g.

````ruby
def initialize(numeral)
  @number = validate(numeral)
end
````
And have the `validate` method handle all validation logic (including `upcase`).